stream_writer.go

```diff
@@ -117,7 +125,10 @@ func (sw *StreamWriter) PrepareIncremental() error {
 		return nil
 	}
 	if sw.prevLevel == 0 {
-		return fmt.Errorf("Unable to do incremental writes because L0 has data")
+		if err := sw.db.Flatten(3); err != nil {
```

Comment:
What does this do? Can you write a comment about how we used to do it before, and why its okay now to flatten instead of cancelling here?

Reply:
Done.